near-sdk-macros/src/lib.rs

```diff
@@ -80,6 +80,18 @@ pub fn near_bindgen(_attr: TokenStream, item: TokenStream) -> TokenStream {
         #[cfg(feature = "__abi-generate")]
         let abi_generated = abi::generate(&item_impl_info);
 
+        for method in &item_impl_info.methods {
+            if method.attr_signature_info.ident == "__contract_abi" {
+                return TokenStream::from(
+                    syn::Error::new_spanned(
+                        method.attr_signature_info.original_sig.ident.to_token_stream(),
+                        "use of reserved contract method",
+                    )
+                    .to_compile_error(),
+                );
+            }
+        }
```

Comment:
Looks good, but just noting that this would *technically* be a breaking change. Probably not big enough deal to warrant going into 5.x, but just noting.